src/backend/distributed/commands/table.c

```diff
@@ -507,6 +510,16 @@ PreprocessAlterTableStmt(Node *node, const char *alterTableCommand,
 		return PreprocessAlterTableStmt(node, alterTableCommand, processUtilityContext);
 	}
 
+	if (AlterTableDropsForeignKey(alterTableStatement))
```

Comment:
I'm lost a little. Why did we move this check to here from `if (!OidIsValid(leftRelationId)) `?

Reply:
Ah, sorry misread